### DISCUSSION TITLE
Restart application

### DISCUSSION THREAD

**Discussion by karpovv-boris** [ShankarBUS/ModernFlyouts#514] [Q&A]
![button-2](https://user-images.githubusercontent.com/78983523/111616450-f52af280-87f2-11eb-9cc5-9790b98288d2.png)

I want this button [restart] in settings (like that btn in PowerToys).
So, I started with introduction separated process, two-way pipe and bla-bla-bla and.. yeap `AppLifecycleManager` already has it.
Now I try reuse it. So, my question is, in case of using `RegisterApplicationRestart(...)` how to properly exit from wpf-app(?) ?

I mean, could we use this callback somehow from `FlyoutHandler.CreateWndProc()`?
![code](https://user-images.githubusercontent.com/78983523/111616990-9d40bb80-87f3-11eb-8281-2e5160303da4.png)

**Comment by Samuel12321** [comment]
Perhaps turn this into an issue - it's easier to track.

**Comment by prakharb5** [comment]
Now tracked in #521.

**Reply by karpovv-boris** [reply]
Thanks!
